### PR TITLE
EVAKA-4176: display new service needs to admin role

### PR DESCRIPTION
**Pull request by Joosakur**
Screenshot 2021-04-15 at 12 48 52

### CODE REVIEW

File: frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                  {sn.startDate.format()} - {sn.endDate.format()}
+                </Td>
+                <Td>{sn.option.name}</Td>
+                <Td>{sn.shiftCare ? i18n.common.yes : i18n.common.no}</Td>
+                <Td>
+                  <StatusLabel
+                    status={getStatusLabelByDateRange({
+                      startDate: sn.startDate,
+                      endDate: sn.endDate
+                    })}
```
Comment (jaahon) [inline frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx:51]: You can just do `getStatusLabelByDateRange(sn)`.

File: frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import { Table, Tbody, Td, Th, Thead, Tr } from 'lib-components/layout/Table'
+import { H4 } from 'lib-components/typography'
+import React from 'react'
+import { NewServiceNeed } from '../../../types/child'
+import _ from 'lodash'
+import StatusLabel from '../../common/StatusLabel'
+import { getStatusLabelByDateRange } from '../../../utils/date'
+import styled from 'styled-components'
+import { defaultMargins } from '../../../../lib-components/white-space'
```
Comment (jaahon) [inline frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx:13]: Use `... from 'lib-components/white-space'`.

File: frontend/src/employee-frontend/components/child-information/placements/PlacementRow.tsx [hunk 3]
```diff
@@ -289,6 +291,12 @@ function PlacementRow({ placement, onRefreshNeeded, checkOverlaps }: Props) {
             </FixedSpaceRow>
           </ActionRow>
         )}
+
+        <RequireRole oneOf={['ADMIN']}>
+          <Gap size="s" />
+
+          <NewServiceNeeds serviceNeeds={placement.serviceNeeds} />
+        </RequireRole>
```
Comment (jaahon) [inline frontend/src/employee-frontend/components/child-information/placements/PlacementRow.tsx:299]: I would hide this in production at least for now.

File: service/src/main/kotlin/fi/espoo/evaka/serviceneednew/NewServiceNeedQueries.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+): List<NewServiceNeed> {
+    // language=SQL
+    val sql =
+        """
+        SELECT sn.id, sn.placement_id, sn.start_date, sn.end_date, sn.shift_care, sno.id as option_id, sno.name as option_name
+        FROM new_service_need sn
+        JOIN service_need_option sno on sno.id = sn.option_id
+        JOIN placement pl ON pl.id = sn.placement_id
+        WHERE pl.unit_id = :unitId
+            AND ((:start IS NULL AND :end IS NULL) OR daterange(:start, :end, '[]') && daterange(sn.start_date, sn.end_date, '[]'))
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/serviceneednew/NewServiceNeedQueries.kt:41]: This `(:start IS NULL AND :end IS NULL) OR ` condition is unnecessary.

File: frontend/src/employee-frontend/components/child-information/placements/PlacementRow.tsx (hunk 3 above)
Comment (Joosakur) [reply frontend/src/employee-frontend/components/child-information/placements/PlacementRow.tsx]: done

File: service/src/main/kotlin/fi/espoo/evaka/serviceneednew/NewServiceNeedQueries.kt (hunk 4 above)
Comment (Joosakur) [reply service/src/main/kotlin/fi/espoo/evaka/serviceneednew/NewServiceNeedQueries.kt]: removed

File: frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx (hunk 1 above)
Comment (Joosakur) [reply frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx]: changed 👍

**Review by jaahon — APPROVED** [review]
Good stuff 👍

**Pull request merged**